### PR TITLE
Store latitude and longitude coordinates

### PR DESCRIPTION
**Pull request by Monica-Coira**
## Description

- Creates an input field for user's college/university and saves this data in the database.
- Gets, stores, and imports my Google Maps API Key.
- Implements a Place Autocomplete Address Form using Google Maps API to allow users to input the beginning of their address and then choose from a dropdown of valid Google Maps addresses.
- Utilizes the Places Library from Google Maps API to get the latitude and longitude coordinates of the user's address and stores this information in the database.
- In the next PR, I will use these latitude and longitude coordinates to calculate location compatibility between users. 

## Milestones
This works towards Milestone 7, which is that users can be matched into study groups based on a compatibility score (technical challenge 1).

## Resources
I used Google Maps API documentation (specifically about the Place Autocomplete Address Form) to implement my address input: [Google Maps API Documentation](https://developers.google.com/maps/documentation/javascript/examples/places-autocomplete-addressform#maps_places_autocomplete_addressform-javascript).

## Test Plan

https://github.com/user-attachments/assets/7eab04cd-69df-470e-a4c7-e576f87ae09a

Screenshot 2025-07-14 at 11 04 42 AM
Besides testing basic functionality, these are the corner cases I tested:

- Attempted to submit the signup form without inputting an address and university to ensure that an error message appears.
- Changed the address multiple times before submitting the signup form to ensure that the latitude and longitude coordinates are only stored for the updated address.
- Verified that the correct longitude and latitude coordinates for each address are stored in the database using Google Maps.

### CODE REVIEW

File: server/routes/auth.js [hunk 1]
```diff
@@ -6,10 +6,10 @@ const prisma = new PrismaClient();
 
 // Create an account
 router.post('/signup', async (req, res) => {
-    const {name, username, password, preferred_start_time, preferred_end_time} = req.body
+    const {name, username, password, preferred_start_time, preferred_end_time, school, latitude, longitude} = req.body
     try {
-        if (!username || !password || !name || !preferred_start_time || !preferred_end_time) {
-            return res.status(400).json({error: "Username, password, name, and preferences are required."})
+        if (!username || !password || !name || !preferred_start_time || !preferred_end_time || !school || !latitude || !longitude) {
```
Comment (alisahq) [inline server/routes/auth.js:11]: If we have one factor/variable missing, wonder if we could still do the matching algorithm and compute the similarity score?

File: client/src/components/AutocompleteAddress.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        }
+        if (place.geometry){
+            const latitude = place.geometry.location.lat();
+            const longitude = place.geometry.location.lng();
+            addAddressCoordinates(latitude, longitude);
+        }
+    }
+
+    const autocompleteInput = () => {
+        const autocomplete = new window.google.maps.places.Autocomplete(inputRef.current, {
```
[the remark this answers is not in the document]
Comment (Monica-Coira) [reply client/src/components/AutocompleteAddress.jsx:21]: Yes, it's the Google Maps JavaScript API that uses the Places Library to provide an autocomplete feature when users are typing in their address:
Screenshot 2025-07-15 at 11 27 23 AM
I also use this API to get the latitude and longitude coordinates for each address.

File: server/routes/auth.js (hunk 1 above)
Comment (Monica-Coira) [reply server/routes/auth.js:11]: Got it! Just implemented this in a new PR: #28. If there are factors/variables missing, the weights for the available variables are redistributed accordingly so that the sum of the weights will always equal 1.
